### ISSUE TITLE
Is leaves even a vanilla + bukkit plugin API and nothing else equivalent?

### ISSUE THREAD

**Issue by mhtvsSFrpHdE** [LeavesMC/Leaves#336]
I've discovered config like
```
redstone-shears-wrench: true
```
Enabled **BY DEFAULT** in `leaves.yml`.
I created a Java 1.21.1 single play save and use shears right click on redstone repeater, it didn't rotate. So it is not a vanilla feature.

It says, "Fork of Paper aimed at repairing broken vanilla properties."  
But yet another fork, you need to **MANUALLY** tweak tons of properties to make it "unoptimized" to function like vanilla?  
For example, paper breaking things for sure, and leaves may add things non exists in vanilla, the result would just the two are going in different direction, but generally speaking, they are actually both break.  
How many other hidden things are enabled silently, and you can never revert it back, because you're unsure is this feature provided in vanilla or not.

And for Spigot...they skip ticks.  
For CraftBukkit, plugins claims support bukkit doesn't support CraftBukkit because they assume people use Spigot or Paper.  
For leaves (Still evaluating), I get a strange wrench and fake players (what? Integrate plugin features into server itself?)

Anyone here can point out what is the current state of survival mode + redstone when it comes to server side?  
Does it indeed required to drop entire bukkit ecosystem like Residence CoreProtect dynmap multiverse and go to something like Fabric to make redstone function properly but don't add unwanted mods?

**Comment by GLDYM** [comment]
Repairing broken vanilla properties **DOESNOT** means making a vanilla + bukkit plugin API server. The aim of Leaves is making a fabric-carpet like + bukkit plugin API server. To do this, we must repair these broken vanilla properties.

---

## if you want it then you have to take it by yourself.

**Comment by s-yh-china** [comment]
I'm not sure what you're talking about
Firstly, the plugins you mentioned can always be used on Leaves.
And why issue?



---Original---
From: ***@***.***&gt;
Date: Tue, Sep 10, 2024 08:50 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [LeavesMC/Leaves] Is leaves even a vanilla + bukkit plugin API and nothing else equivalent? (Issue #336)




 
I've discovered config like
 redstone-shears-wrench: true  
Enabled BY DEFAULT in leaves.yml.
 I created a Java 1.21.1 single play save and use shears right click on redstone repeater, it didn't rotate. So it is not a vanilla feature.
 
It says, "Fork of Paper aimed at repairing broken vanilla properties."
 But yet another fork, you need to MANUALLY tweak tons of properties to make it "unoptimized" to function like vanilla?
 For example, paper breaking things for sure, and leaves may add things non exists in vanilla, the result would just the two are going in different direction, but generally speaking, they are actually both break.
 How many other hidden things are enabled silently, and you can never revert it back, because you're unsure is this feature provided in vanilla or not.
 
And for Spigot...they skip ticks.
 For CraftBukkit, plugins claims support bukkit doesn't support CraftBukkit because they assume people use Spigot or Paper.
 For leaves (Still evaluating), I get a strange wrench and fake players (what? Integrate plugin features into server itself?)
 
Anyone here can point out what is the current state of survival mode + redstone when it comes to server side?
 Does it indeed required to drop entire bukkit ecosystem like Residence CoreProtect dynmap multiverse and go to something like Fabric to make redstone function properly but don't add unwanted mods?

**Comment by mhtvsSFrpHdE** [comment]
> Repairing broken vanilla properties **DOESNOT** means making a vanilla + bukkit plugin API server. The aim of Leaves is making a fabric-carpet like + bukkit plugin API server. To do this, we must repair these broken vanilla properties.
> ## if you want it then you have to take it by yourself.

Had a quick shot on carpet mod and understand its feature. Thank you for your time.

**Comment by Lumine1909** [comment]
This server project is aimed for technical survival servers, almost all the features are configurable (besides shears wrench and fakeplayer).
The feature you pointed out is a simple and it’s convenient for the survival mode redstone user. So we marked it true as default.
In addition, you can find all the features and descriptions in our manual, I don’t believe that a server owner who owns a technical server is not able to read a manual or ask questions in our discord before it turn into this server project.
In summary, this issue is considered meaningless. If you want to contribute to this project, please fix some bugs instead of creating meaningless issues.
If you still want to change the things you pointed, you can just fork this project and change it yourself.
Thank you!

**Comment by Lumine1909** [comment]
In addition, it seems you want to decide fabric server or leaves server for your minecraft server.
In my opinion, this server is not your best choice unless you want vanilla + technical survival mod support.
If you still have some questions, feel free to join our discord.

**Comment by s-yh-china** [comment]
vanilla + technical survival mod support + bukkit api
If only the first two are needed, why not fabric

**Comment by MC-XiaoHei** [comment]
The design philosophy of leaves is closer to enhancing the TMC player experience than just guaranteeing vanilla features, i think

**Comment by mhtvsSFrpHdE** [comment]
> This server project is aimed for technical survival servers, almost all the features are configurable (besides shears wrench and fakeplayer). The feature you pointed out is a simple and it’s convenient for the survival mode redstone user. So we marked it true as default. In addition, you can find all the features and descriptions in our manual, I don’t believe that a server owner who owns a technical server is not able to read a manual or ask questions in our discord before it turn into this server project. In summary, this issue is considered meaningless. If you want to contribute to this project, please fix some bugs instead of creating meaningless issues. If you still want to change the things you pointed, you can just fork this project and change it yourself. Thank you!

Please delete this issue.

Until someone build a redstone tick precise structure in my server and ask why it doesn't work: it will stop after a while. I don't know what are those pistons used for, but they are indeed will cycle the one work in vanilla MC (Singleplayer). I realize there is indeed something wrong with spigot.

Then I go from comment of [Spigot/PaperMC is NOT Vanilla and it's broken. Here's why | Minecraft Discussions](https://www.youtube.com/watch?v=EEr1Ebxa7J8), people mention "There's a fork of paper that fixes broken vanilla properties called Leaves" and a reply "You are an absolute legend, thank you so much", these comments make me to think leaves will be the final solution.

I look at https://leavesmc.org and playaround with yml and https://docs.leavesmc.org/en/leaves/reference/configuration  
Soon realized it has a tendency of more like a modpack instead of "fixes broken" only.

"aimed at repairing broken vanilla properties" continuously draws my attention and finally mislead me to post things to ask. I didn't notice any message on leavesmc.org or docs.leavesmc.org or https://github.com/LeavesMC/Leaves says "Don't post support on GitHub, go to Discord instead". I've just searched a few keywords in leaves GitHub issue to see if I can find something, so the last website on my browser is GitHub, and I created the issue. My behavior seems to be treated as spamming git issue, I'm sorry for that.

> if you want it then you have to take it by yourself.

As I understand more and more about paper, I found a page on paper website: [Vanilla-like Experience](https://docs.papermc.io/paper/vanilla)  
It also mentioned several spigot.yml change I can apply to my server, and another page says: [Spigot Tick Limiter: Don’t use “max-tick-time”!](https://aikar.co/2015/10/08/spigot-tick-limiter-dont-use-max-tick-time) I also modified the two value to 1000.  
Finally, I installed [TabTPS](https://www.spigotmc.org/resources/tabtps-1-8-8-1-21-show-tps-mspt-and-more-in-the-tab-menu.82528) on my server and expose tps and mstp permission to player.  
I can't really throw out the entire bukkit ecosystem in the end and decide not to go with Fabric.  
In Fabric, I can do anything I have currently done, just in a different and I don't familiar way.

All of these are start by a player's support DM. I've had enough internet for today and exhausted on this topic.
